Sprint-1/destructuring/exercise-2/exercise.js

```diff
@@ -70,3 +70,44 @@ let hogwarts = [
     occupation: "Teacher",
   },
 ];
+
+
+//1
+// const newItem = oldItem.filter(({ propWeWant }) => propWeWant === 'propValue')
+
+const sortingHat = (hogwarts) => {
+  // filter gryffindor
+  const filtered = hogwarts.filter(({ house }) => house === "Gryffindor");
+
+  // return names
+  let gryffindor = [];
+  filtered.forEach(({ firstName, lastName }) => {
+    gryffindor.push(`${firstName} ${lastName}`);
+  })
+
+  return gryffindor;
+}
+  
+//2
+// const newItem = oldItem.filter(({ propWeWant1, propWeWant2 }) => propWeWant 1 === 'propValue' && propWeWant2 === 'propValue2')
+// or if checking if prop exists
+// const newItem = oldItem.filter(({ propWeWant1, propWeWant2 }) => propWeWant 1 === 'propValue' && propWeWant2)
+const findTeachersWithPets = (hogwarts) => {
+  let teachersWithPets = [];
+
+  // filter teachers
+  // const filtered = hogwarts.filter(({ occupation, pet }) => occupation === 'Teacher');
+
+  // check for pets
+  // const withPets = filtered.filter(({ pet }) => pet));
+
+//filter for teachers with pets
+  const filtered = hogwarts.filter(({ occupation, pet }) => occupation === 'Teacher' && pet);
+
+
+  filtered.forEach(({ firstName, lastName }) => {
+    teachersWithPets.push(`${firstName} ${lastName}`);
+  })
+
+  return teachersWithPets;
+}
```

Comment:
When I add the lines:

```js
console.log(sortingHat(hogwarts));
console.log(findTeachersWithPets(hogwarts));
```

I do get the expected output when running the file. You've used destructuring effectively here to pick out from objects only the specific fields that you need. Good stuff!